src/components/Upload.tsx

```diff
@@ -1,9 +1,11 @@
+import DataTable from './DataTable';
 import NavigationButton from './NavigationButton';
 
 function Upload() {
   return (
     <div className="flex flex-col items-center">
       <h1>Upload</h1>
+      <DataTable />
```

Comment:
I like the modular approach, but as I said in the other component, we should split the "upload" and "display" concerns between the Upload and the DataTable component. e.g. I would expect to look at this component to see both the upload UI elements and the relevant store hooks to upload data.